### ISSUE TITLE
Docker Desktop 3.3.1 not starting on Mac M1

### ISSUE THREAD

**Issue by mohsinht** [docker/for-mac#5602]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 4E395100-2AE9-4F41-9F5E-5228C1979B6F/20210419181033

### Expected behavior
The docker desktop must start.
### Actual behavior
The icon shows an opening animation, I get a small docker icon on the top bar, but clicking on docker desktop does not start the app.
### Information

  - macOS Version: Big Sur 11.2.3
  - Intel chip or Apple chip: Apple M1
  - Docker Desktop Version: 3.3.1

### Steps to reproduce the behavior

  1. Uninstall previous installations and then download from here https://docs.docker.com/docker-for-mac/apple-silicon/
  2. Install and then drag the app icon in applications.
  3. Run the app

**Comment by djs55** [comment]
@mohsinht thanks for your report. Could you reproduce the problem and upload diagnostics and quote the ID here? If the UI is responding you can use: the whale menu -> Troubleshoot -> Get Support -> Upload diagnostics. If the UI isn't responding then you can run
```
/Applications/Docker.app/Contents/MacOS/com.docker.diagnose gather -upload
```

**Comment by trevorrobertsjr** [comment]
I am experiencing the same behavior. Here is my Diagnostics ID: 4E395100-2AE9-4F41-9F5E-5228C1979B6F/20210419181033

**Comment by denisviklov** [comment]
The same here but I've already remove everything

**Comment by mohsinht** [comment]
@djs55  I cannot provide the diagnostics because I downgraded my setup to 3.3.0 (62632) as the latest one wasn't working. I do not want to break it again so waiting for a fix.

The diagnostics mentioned in the above comment by @trevorrobertsjr should be same as mine if he's using Apple Silicon too.

**Comment by djs55** [comment]
@trevorrobertsjr thanks for the diagnostics. I notice that your system is using the `virtualization.framework` and you seem to have no network connectivity between the VM and the host. This is often caused by VPN or corporate security software.

Could you disable `virtualization.framework` by editing `~/Library/Group Containers/group.com.docker/settings.json` and set `useVirtualizationFramework` to `false`?

**Comment by sawyerh** [comment]
I was having the same issue of Docker Desktop 3.3.3 not starting on my M1. I followed @djs55's suggestion above to disable `virtualization.framework`. Docker Desktop successfully started up after making this change and restarting my computer.

**Comment by trevorrobertsjr** [comment]
This fixed the problem! Thank you @djs55!!!

**Comment by djs55** [comment]
I think this issue should be fixed in Docker Desktop 3.4.0, both when `virtualization.framework` is enabled and disabled.

Thanks for the report!
